protocols/identify/src/behaviour.rs

```diff
@@ -357,7 +376,9 @@ impl NetworkBehaviour for Behaviour {
             self.events.extend(change_events)
         }
 
-        if listen_addr_changed && self.config.push_listen_addr_updates {
+        if (listen_addr_changed && self.config.push_listen_addr_updates)
+            | (external_addr_changed && self.config.push_external_addr_updates)
```

Comment:
Why use a bit-wise OR here instead of a logical one?

Reply:
Most certainly a typo